### ISSUE TITLE
unreadable_literal is annoying for hex numbers

### ISSUE THREAD

**Issue by vks** [rust-lang-nursery/rust-clippy#2842]
When having hex literals in code, conciseness is usually more important than readability, so the lint seems to be counter-productive in such cases.

**Comment by flip1995** [comment]
Do you have some examples for this? IMO this lint suggests more readable Hex-literals (most of the time).

**Comment by newpavlov** [comment]
For example in RustCrypto crates we selectively disable `unreadable_literal` for most of the constants. I just don't see point in making files like [this](https://github.com/RustCrypto/hashes/blob/b0d27c06634fa1950bc2aae5ffcf3b7f323b918f/streebog/src/table.rs) more "readable".

**Comment by flip1995** [comment]
Yeah making walls of hex-literals more readable doesn't really make sense. But I would say this is an edge case, since you shouldn't have things like this spread around in your code, but only in a few special files. Disabling the lint in those files seems reasonable to me.

This lint aims more on single literals in your Code, like
```rust
const IM_A_FLAG: u16 = 0x00FF8000;
vs
const IM_A_READABLE_FLAG: u16 = 0x00FF_8000;
```

IMO the second version is nicer.
